client/consensus/relay-chain/src/lib.rs

```diff
@@ -224,7 +230,25 @@ where
 			return None;
 		}
 
-		Some(ParachainCandidate { block, proof })
+		// This DOES print, so our own import is working
+		println!("Debug info: we made it here");
+
+
+		// My question is about this part. If I don't include this, my own node imports the block,
+		// but cannot retrieve the state, because it has the wrong block hash.
+		//
+		// But if I leave this in place, then the relay chain cannot validate the candidate.
+		// It fails with:
+		// 2021-03-03 15:58:54  panicked at 'assertion failed: `(left == right)`
+		// left: `1`,
+		// right: `0`: Number of digest items must match that calculated.', /home/joshy/.cargo/git/checkouts/substrate-7e08433d4c370a21/74d5612/frame/executive/src/lib.rs:421:9
```

Comment:
You need to remove the digest before importing the block. Check how Babe and AuRa are doing this.

Reply:
Thanks for taking a look. I understand that I need to remove the digest before executing the block. I'm not sure where that is expected to happen with the current parachain  consensus trait.

It's the relay chain validators that are giving this error when they execute the block. Should produce candidate return the block with the seal digest or without it?

Reply:
To say a little more. I know my verifier will have to pop the seal off in order for other parachain nodes to import this block. But validators won't even run my verifier. So the verifier alone will not solve this problem.

Reply:
https://github.com/paritytech/cumulus/blob/8c1f788aa34ea5213544f813903a347d33ce6d47/rococo-parachains/runtime/src/lib.rs#L441

https://github.com/paritytech/cumulus/blob/8c1f788aa34ea5213544f813903a347d33ce6d47/pallets/aura/src/lib.rs#L53-L99

;)

Reply:
:exploding_head: :pray: